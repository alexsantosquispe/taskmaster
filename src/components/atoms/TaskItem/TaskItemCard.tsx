import { Avatar } from '../Avatar/Avatar';
import { Chip } from '../Chip/Chip';
import { DateBadge } from '../DateBadge/DateBadge';
import { ProgressBadge } from '../ProgressBadge/ProgressBadge';
import type { TaskType } from '../../../models/types';
import { getPriorityStyles } from '../../../utils';

type TaskItemCardProps = Omit<TaskType, 'createdAt' | 'status'>;

export const TaskItemCard = ({
  id,
  title,
  description,
  priority,
  progress,
  lastUpdate
}: TaskItemCardProps) => {
  const priorityStyles = getPriorityStyles(priority);
  return (
    <article
      data-id={id}
      className="dark:bg-primary flex w-full flex-col rounded-xl border border-neutral-200 bg-white px-3 text-[0.8125rem] dark:border-neutral-700 dark:text-white"
    >
      <div className="flex flex-col gap-2 pt-2">
        <a
          href="#"
          className="text-[0.9375rem] hover:cursor-pointer hover:text-blue-600 hover:underline dark:hover:text-blue-500"
        >
          {title}
        </a>
        <p className="line-clamp-2 max-h-12 text-neutral-700 dark:text-white/80">
          {description}
        </p>
      </div>

      <div className="pt-2">
        <Chip label={priority} className={priorityStyles} />
      </div>

      <div className="flex w-full items-center justify-between py-2 text-neutral-600">
        <DateBadge date={lastUpdate} />
        <ProgressBadge percentage={progress} />
      </div>

      <hr className="text-neutral-200 dark:text-neutral-700" />
      <div className="flex items-center justify-between gap-2 py-2 dark:text-white/80">
        <Avatar />
        <div>comments section</div>
      </div>
    </article>
  );
};
