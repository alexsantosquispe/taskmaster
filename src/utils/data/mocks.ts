import {
  PRIORITY_TYPES,
  STATUS_TYPES,
  type TaskType
} from '../../models/types';

export const BACKLOG_TASKS: TaskType[] = [
  {
    id: '1',
    title: 'Set up analytics tools',
    description:
      'Install and configure Google Analytics and Hotjar to monitor user behavior and traffic.',
    status: STATUS_TYPES.BACKLOG,
    createdAt: '2025-07-01T10:24:00-04:00',
    lastUpdate: '2025-07-05T14:13:00-04:00',
    priority: PRIORITY_TYPES.MEDIUM,
    progress: 0
  },
  {
    id: '2',
    title: 'Gather user feedback on onboarding',
    description:
      'Collect and analyze user feedback from early access users about the onboarding experience.',
    status: STATUS_TYPES.BACKLOG,
    createdAt: '2025-07-02T11:30:00-04:00',
    lastUpdate: '2025-07-06T09:00:00-04:00',
    priority: PRIORITY_TYPES.HIGH,
    progress: 0
  },
  {
    id: '18',
    title: 'Write project README',
    description:
      'Added complete project documentation including setup, scripts, and folder structure.',
    status: STATUS_TYPES.DONE,
    createdAt: '2025-07-03T13:40:00-04:00',
    lastUpdate: '2025-07-06T10:00:00-04:00',
    priority: PRIORITY_TYPES.LOW,
    progress: 100
  }
];

export const ON_HOLD_TASKS: TaskType[] = [
  {
    id: '7',
    title: 'Finalize pricing model options',
    description:
      'Pricing structure needs stakeholder review before publishing. Waiting for finance team input.',
    status: STATUS_TYPES.ON_HOLD,
    createdAt: '2025-07-03T10:00:00-04:00',
    lastUpdate: '2025-07-09T11:45:00-04:00',
    priority: PRIORITY_TYPES.LOW,
    progress: 75
  }
];

export const IN_PROGRESS_TASKS: TaskType[] = [
  {
    id: '12',
    title: 'Implement authentication flow',
    description:
      'OAuth integration is partially done; currently wiring up token refresh and user session handling.',
    status: STATUS_TYPES.IN_PROGRESS,
    createdAt: '2025-07-06T09:30:00-04:00',
    lastUpdate: '2025-07-10T10:45:00-04:00',
    priority: PRIORITY_TYPES.HIGH,
    progress: 45
  },
  {
    id: '13',
    title: 'Integrate Supabase for data storage',
    description:
      'Connection established. Working on CRUD operations and testing permissions for different roles.',
    status: STATUS_TYPES.IN_PROGRESS,
    createdAt: '2025-07-08T14:00:00-04:00',
    lastUpdate: '2025-07-10T09:30:00-04:00',
    priority: PRIORITY_TYPES.HIGH,
    progress: 80
  }
];

export const REVIEW_TASKS: TaskType[] = [
  {
    id: '8',
    title: 'Update brand guidelines',
    description:
      'Rebranding paused until feedback from the design lead is received.',
    status: STATUS_TYPES.REVIEW,
    createdAt: '2025-07-02T14:20:00-04:00',
    lastUpdate: '2025-07-06T17:00:00-04:00',
    priority: PRIORITY_TYPES.MEDIUM,
    progress: 100
  },
  {
    id: '9',
    title: 'Legal review for terms of service',
    description:
      'Sent draft to legal team. Awaiting confirmation before moving forward.',
    status: STATUS_TYPES.REVIEW,
    createdAt: '2025-07-05T13:10:00-04:00',
    lastUpdate: '2025-07-08T12:30:00-04:00',
    priority: PRIORITY_TYPES.HIGH,
    progress: 100
  },
  {
    id: '19',
    title: 'Create logo and favicon',
    description:
      'Exported multiple sizes and formats for the logo. Favicon added to HTML head.',
    status: STATUS_TYPES.DONE,
    createdAt: '2025-07-04T08:00:00-04:00',
    lastUpdate: '2025-07-07T09:20:00-04:00',
    priority: PRIORITY_TYPES.MEDIUM,
    progress: 100
  }
];

export const DONE_TASKS: TaskType[] = [
  {
    id: '15',
    title: 'Deploy staging environment',
    description:
      'Staging environment successfully deployed and tested with mock data.',
    status: STATUS_TYPES.DONE,
    createdAt: '2025-06-30T11:00:00-04:00',
    lastUpdate: '2025-07-03T15:30:00-04:00',
    priority: PRIORITY_TYPES.HIGH,
    progress: 100
  },
  {
    id: '16',
    title: 'Fix navbar responsiveness',
    description:
      'Navbar layout now works correctly across all screen sizes and breakpoints.',
    status: STATUS_TYPES.DONE,
    createdAt: '2025-07-01T10:00:00-04:00',
    lastUpdate: '2025-07-04T12:15:00-04:00',
    priority: PRIORITY_TYPES.MEDIUM,
    progress: 100
  },
  {
    id: '17',
    title: 'Set up CI/CD pipeline',
    description:
      'GitHub Actions configured to run tests and deploy to Vercel on push to main.',
    status: STATUS_TYPES.DONE,
    createdAt: '2025-07-02T09:30:00-04:00',
    lastUpdate: '2025-07-05T14:45:00-04:00',
    priority: PRIORITY_TYPES.HIGH,
    progress: 100
  },

  {
    id: '20',
    title: 'Test form validation logic',
    description:
      'Test cases written and validated for all form fields using React Hook Form.',
    status: STATUS_TYPES.DONE,
    createdAt: '2025-07-05T10:45:00-04:00',
    lastUpdate: '2025-07-08T11:30:00-04:00',
    priority: PRIORITY_TYPES.HIGH,
    progress: 100
  }
];
