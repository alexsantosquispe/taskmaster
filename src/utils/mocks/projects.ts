import type { ProjectDTO } from '../../services/apiTypes';

export const PROJECTS: ProjectDTO[] = [
  {
    id: '4e393d82-2ec9-46e6-89a1-f42aca499e24',
    name: 'Spotify Clone App',
    description: 'This is new project with react native',
    create_date: '2025-08-22T17:24:45.746846+00:00',
    color: '#2563eb',
    code: 'SCA',
    update_date: '2025-08-23T20:00:00.746846+00:00'
  },
  {
    id: 'ac2eed41-680b-4cd9-b079-7db1e4fac09e',
    name: 'Resume Builder V1',
    description:
      'Tihs project is a web app to create an resume and check the how it can be improved with AI',
    create_date: '2025-08-21T03:59:50.116667+00:00',
    color: '#f97316',
    code: 'RESB',
    update_date: '2025-08-21T03:59:50.116667+00:00'
  },
  {
    id: 'd9587eaa-1000-49db-8ce9-183fa47a1696',
    name: 'Task Master Mobile App',
    description: 'This is a mobile version with react native',
    create_date: '2025-08-21T03:40:21.137062+00:00',
    color: '#eab308',
    code: 'TMMA',
    update_date: '2025-08-21T03:40:21.137062+00:00'
  },
  {
    id: '5082b2a0-1e2a-43ca-9661-a12260520a15',
    name: 'Web Track App',
    description: 'Agile project management tool',
    create_date: '2025-08-18T23:51:48.391714+00:00',
    color: '#8B5CF6',
    code: 'DEV',
    update_date: '2025-08-18T23:51:48.391714+00:00'
  }
];
