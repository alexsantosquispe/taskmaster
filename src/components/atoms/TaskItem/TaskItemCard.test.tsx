import { render } from '@testing-library/react';
import { BACKLOG_TASKS } from '../../../utils/data/mocks';
import { TaskItemCard, type TaskItemCardProps } from './TaskItemCard';

describe('TaskItemCard', () => {
  const props: TaskItemCardProps = BACKLOG_TASKS[0];
  it('should render the component correctly', () => {
    const component = render(<TaskItemCard {...props} />);

    expect(component).toMatchSnapshot();
  });
});
