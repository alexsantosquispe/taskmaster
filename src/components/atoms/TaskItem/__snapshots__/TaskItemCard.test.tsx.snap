// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`TaskItemCard should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="flex w-full flex-col rounded-xl border border-neutral-200 bg-white px-3 text-[0.8125rem] dark:border-neutral-700 dark:bg-transparent dark:text-white"
        data-id="1"
      >
        <div
          class="flex flex-col gap-2 pt-2"
        >
          <a
            class="text-[0.9375rem] hover:cursor-pointer hover:text-blue-600 hover:underline dark:hover:text-orange-500"
            href="#"
          >
            Set up analytics tools
          </a>
          <p
            class="line-clamp-2 max-h-12 text-neutral-700 dark:text-white/80"
          >
            Install and configure Google Analytics and Hotjar to monitor user behavior and traffic.
          </p>
        </div>
        <div
          class="pt-2"
        >
          <div
            class="flex w-fit min-w-13 items-center justify-center gap-2 rounded-xl px-2 py-0.5 text-xs font-medium bg-orange-100 text-amber-800 border border-amber-800 dark:bg-orange-600/20 dark:border-orange-400 dark:text-orange-400"
          >
            <span>
              Medium
            </span>
          </div>
        </div>
        <div
          class="flex w-full items-center justify-between py-2 text-neutral-600"
        >
          <div
            class="flex items-center gap-2 text-neutral-600 dark:text-white/80"
          >
            <svg
              class="size-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 2v4"
              />
              <path
                d="M16 2v4"
              />
              <rect
                height="18"
                rx="2"
                width="18"
                x="3"
                y="4"
              />
              <path
                d="M3 10h18"
              />
              <path
                d="M8 14h.01"
              />
              <path
                d="M12 14h.01"
              />
              <path
                d="M16 14h.01"
              />
              <path
                d="M8 18h.01"
              />
              <path
                d="M12 18h.01"
              />
              <path
                d="M16 18h.01"
              />
            </svg>
            <span>
              Jul 5
            </span>
          </div>
          <div
            class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
          >
            <svg
              height="16"
              width="16"
            >
              <circle
                class="stroke-gray-200 dark:stroke-neutral-600"
                cx="8"
                cy="8"
                fill="none"
                r="7"
                stroke-width="2"
              />
              <circle
                class="stroke-primary dark:stroke-white text-neutral-600 dark:text-white/80"
                cx="8"
                cy="8"
                fill="none"
                r="7"
                stroke-dasharray="43.982297150257104"
                stroke-dashoffset="43.54247417875453"
                stroke-linecap="round"
                stroke-width="2"
                transform="rotate(-90 8 8)"
              />
            </svg>
            <span
              class="stroke-primary dark:stroke-white text-neutral-600 dark:text-white/80"
            >
              0
              %
            </span>
          </div>
        </div>
        <hr
          class="text-neutral-200 dark:text-neutral-700"
        />
        <div
          class="flex items-center justify-between gap-2 py-2 dark:text-white/80"
        >
          <div
            class="flex w-fit items-center justify-center overflow-hidden rounded-full bg-gray-200 p-0.5 text-neutral-500"
          >
            <img
              alt="User avatar image"
              class="h-5 w-5"
              src="/img/user-avatar.webp"
            />
          </div>
          <div
            class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
          >
            <svg
              class="size-4 text-neutral-400"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.9 20A9 9 0 1 0 4 16.1L2 22Z"
              />
              <path
                d="M8 12h.01"
              />
              <path
                d="M12 12h.01"
              />
              <path
                d="M16 12h.01"
              />
            </svg>
            <span>
              23
            </span>
          </div>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="flex w-full flex-col rounded-xl border border-neutral-200 bg-white px-3 text-[0.8125rem] dark:border-neutral-700 dark:bg-transparent dark:text-white"
      data-id="1"
    >
      <div
        class="flex flex-col gap-2 pt-2"
      >
        <a
          class="text-[0.9375rem] hover:cursor-pointer hover:text-blue-600 hover:underline dark:hover:text-orange-500"
          href="#"
        >
          Set up analytics tools
        </a>
        <p
          class="line-clamp-2 max-h-12 text-neutral-700 dark:text-white/80"
        >
          Install and configure Google Analytics and Hotjar to monitor user behavior and traffic.
        </p>
      </div>
      <div
        class="pt-2"
      >
        <div
          class="flex w-fit min-w-13 items-center justify-center gap-2 rounded-xl px-2 py-0.5 text-xs font-medium bg-orange-100 text-amber-800 border border-amber-800 dark:bg-orange-600/20 dark:border-orange-400 dark:text-orange-400"
        >
          <span>
            Medium
          </span>
        </div>
      </div>
      <div
        class="flex w-full items-center justify-between py-2 text-neutral-600"
      >
        <div
          class="flex items-center gap-2 text-neutral-600 dark:text-white/80"
        >
          <svg
            class="size-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 2v4"
            />
            <path
              d="M16 2v4"
            />
            <rect
              height="18"
              rx="2"
              width="18"
              x="3"
              y="4"
            />
            <path
              d="M3 10h18"
            />
            <path
              d="M8 14h.01"
            />
            <path
              d="M12 14h.01"
            />
            <path
              d="M16 14h.01"
            />
            <path
              d="M8 18h.01"
            />
            <path
              d="M12 18h.01"
            />
            <path
              d="M16 18h.01"
            />
          </svg>
          <span>
            Jul 5
          </span>
        </div>
        <div
          class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
        >
          <svg
            height="16"
            width="16"
          >
            <circle
              class="stroke-gray-200 dark:stroke-neutral-600"
              cx="8"
              cy="8"
              fill="none"
              r="7"
              stroke-width="2"
            />
            <circle
              class="stroke-primary dark:stroke-white text-neutral-600 dark:text-white/80"
              cx="8"
              cy="8"
              fill="none"
              r="7"
              stroke-dasharray="43.982297150257104"
              stroke-dashoffset="43.54247417875453"
              stroke-linecap="round"
              stroke-width="2"
              transform="rotate(-90 8 8)"
            />
          </svg>
          <span
            class="stroke-primary dark:stroke-white text-neutral-600 dark:text-white/80"
          >
            0
            %
          </span>
        </div>
      </div>
      <hr
        class="text-neutral-200 dark:text-neutral-700"
      />
      <div
        class="flex items-center justify-between gap-2 py-2 dark:text-white/80"
      >
        <div
          class="flex w-fit items-center justify-center overflow-hidden rounded-full bg-gray-200 p-0.5 text-neutral-500"
        >
          <img
            alt="User avatar image"
            class="h-5 w-5"
            src="/img/user-avatar.webp"
          />
        </div>
        <div
          class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
        >
          <svg
            class="size-4 text-neutral-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.9 20A9 9 0 1 0 4 16.1L2 22Z"
            />
            <path
              d="M8 12h.01"
            />
            <path
              d="M12 12h.01"
            />
            <path
              d="M16 12h.01"
            />
          </svg>
          <span>
            23
          </span>
        </div>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
